src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/BreakpointEvidence.java

```diff
@@ -101,6 +102,102 @@ public String stringRep(final ReadMetadata readMetadata, final int minEvidenceMa
         return location.toString() + "\t" + weight + "\t" + this.getClass().getSimpleName() + "\t" + dtString;
     }
 
+    /**
+     * Returns BreakpointEvidence constructed from string representation. Used to reconstruct BreakpointEvidence for
+     * unit tests. It is intende for stringRep() to be an inverse of this function, but not the other way around. i.e.
+     *      fromStringRep(strRep, readMetadata).stringRep(readMetadata, minEvidenceMapQ) == strRep
+     * but it may be the case that
+     *      fromStringRep(evidence.stringRep(readMetadata, minEvidenceMapQ), readMetadata) != evidence
+     */
+    @VisibleForTesting
+    static BreakpointEvidence fromStringRep(final String strRep, final ReadMetadata readMetadata) {
+        final String[] words = strRep.split("\t");
+
+        final SVInterval location = locationFromStringRep(words[0]).getInterval();
+
+        final int weight = Integer.parseInt(words[1]);
+
+        final String evidenceType = words[2];
+        if(evidenceType.equals("TemplateSizeAnomaly")) {
+            final int readCount = Integer.parseInt(words[4]);
+            return new TemplateSizeAnomaly(location, weight, readCount);
+        } else {
+            final List<StrandedInterval> distalTargets = words[3].isEmpty() ? new ArrayList<>()
+                : Arrays.stream(words[3].split(";")).map(BreakpointEvidence::locationFromStringRep)
+                    .collect(Collectors.toList());
+            if(distalTargets.size() > 1) {
+                throw new IllegalArgumentException("BreakpointEvidence must have 0 or 1 distal targets");
+            }
+            final String[] templateParts = words[4].split("/");
+            final String templateName = templateParts[0];
+            final TemplateFragmentOrdinal fragmentOrdinal;
+            if(templateParts.length <= 1) {
+                fragmentOrdinal = TemplateFragmentOrdinal.UNPAIRED;
+            } else switch (templateParts[1]) {
```

Comment:
I think @davidbenjamin was missing a parse call to switch the string to an int, and his suggestion is good. This seems to work:

```
            if(templateParts.length <= 1) {
                fragmentOrdinal = TemplateFragmentOrdinal.UNPAIRED;
            } else {
                fragmentOrdinal = TemplateFragmentOrdinal.values()[Integer.parseInt(templateParts[1])];
            }
```

Reply:
I'd convinced myself indexing into string enums was impossible, but that works. It compiles but it has the additional problem that I'd have to alter the mapping order.
parseValue -> values index
0 -> 4
1 -> 2
2 -> 3
And that leaves "?" needing to be handled by catching an exception (although I don't *think* any BreakpointEvidence uses that now). If there's a concise way to overload / specify an order for the enum, that could work, otherwise I think the switch statement is easiest. If the switch statement is highly undesirable, another possibility could be to build a static map from String to TemplateFragmentOrdinal, and then I could get all the cases right but just call get() from the map.